.github/workflows/ci.yaml

```
@@ -198,3 +198,16 @@ jobs:
             echo '```' >> $GITHUB_STEP_SUMMARY
             exit 1
           fi
+
+  pre-commit:
+    name: "pre-commit"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: "Install Rust toolchain"
+        run: rustup show
+      - run: pip install pre-commit
```

> Can we think of any way to speed that step up? 7m is pretty long.

Adding this snippet should work. Pre-commit caches at `~/.cache/pre-commit` by default; see [pre-commit docs](https://pre-commit.com/#managing-ci-caches).

```suggestion
      - run: pip install pre-commit
      - uses: actions/cache@v3
        with:
            path: ~/.cache/pre-commit
            key: pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}
```